### PR TITLE
Karma-based unit test runner

### PR DESCRIPTION
**Pull request by tailsu**
Integrated with nativescript-cli `test init` and `test run` commands from `sd/karma` branch

### CODE REVIEW

**Comment by enchev** [comment]
run ci

**Comment by enchev** [comment]
Good job @tailsu! Can you please sign our CLA? I don't see also any sample tests, can you add one to know more about how to use your test runner?

Thanks

**Comment by hshristov** [comment]
@tailsu It will be much easier to support this if it was written in TypeScript

**Comment by hdeshev** [comment]
I'll be the neckbeard that asks to convert all tabs to 4-space indents.

tabs

**Comment by tailsu** [comment]
OK, converted everything to TypeScript and fixed spaces.

**Comment by tailsu** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

File: unit-test-runner/main-page.xml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<Page xmlns="http://www.nativescript.org/tns.xsd" loaded="pageLoaded">
+	<GridLayout cssClass="landingContainer">
+		<Image src="~/tns_modules/unit-test-runner/nativescript.png" stretch="none" cssClass="landingBackground"/>
```
Comment (hdeshev) [inline unit-test-runner/main-page.xml]: It seems XML files are still using tabs for indentation.

**Comment by dtopuzov** [comment]
run ci

**Comment by tailsu** [comment]
Extracted unit test runner as npm module `nativescript-unit-test-runner`. Closing this one.

**Pull request closed without merging**